pkg/tombstone/tombstone.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package tombstone
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path"
+	"strconv"
+	"time"
+
+	"github.com/cespare/xxhash"
+	"github.com/go-kit/kit/log"
+	"github.com/prometheus/prometheus/pkg/timestamp"
+
+	"github.com/thanos-io/thanos/pkg/objstore"
+)
+
+const (
+	// TombstoneDir is the name of directory to upload tombstones.
+	TombstoneDir = "tombstones"
+)
+
+// Tombstone represents a tombstone.
+type Tombstone struct {
+	Matcher      string `json:"matcher"`
+	MinTime      int64  `json:"minTime"`
+	MaxTime      int64  `json:"maxTime"`
+	CreationTime int64  `json:"creationTime"`
+}
+
+// NewTombstone returns a new instance of Tombstone.
+func NewTombstone(matcher string, minTime, maxTime int64) Tombstone {
+	return Tombstone{
+		Matcher:      matcher,
+		MinTime:      minTime,
+		MaxTime:      maxTime,
+		CreationTime: timestamp.FromTime(time.Now()),
+	}
+}
+
+// GenName generates file name based on Matcher, MinTime and MaxTime of a tombstone.
+func GenName(ts Tombstone) string {
+	hash := xxhash.Sum64([]byte(ts.Matcher + string(ts.MinTime) + string(ts.MaxTime)))
```

Comment:
interesting (: What's the purpose of this name? Was this logic part of the design?  :thinking: 

The only problem is that it will cause weird errors when someone will try to delete same things. Maybe that's ok - we should at least note it in design or comment here

Reply:
If a same request is repeated in the future, this logic helps overwrite the previous tombstone file and avoids duplicate tombstones. Yeah sure, will add this to the design proposal. :slightly_smiling_face: